playbooks/roles/bifrost-configdrives-dynamic/tasks/main.yml

```diff
@@ -58,23 +50,29 @@
     node_network_data: "{{ node_network_data | default(node_network_info) }}"
   when: addressing_mode is undefined or "dhcp" not in addressing_mode
 
-- name: "Place network data template in each openstack/latest folder"
+- name: "Place network data template in each openstack/<version> folder"
   template:
     src: network_data.json.j2
-    dest: "{{ variable_configdrive_location.stdout }}/{{ uuid }}/openstack/latest/network_data.json"
-  when: addressing_mode is undefined or "dhcp" not in addressing_mode
+    dest: "{{ variable_configdrive_location.stdout }}/{{ uuid }}/openstack/{{ item }}/network_data.json"
+  with_items: "{{ metadata_versions }}"
+  when:
+    - item in metadata_versions_supporting_network_data
+    - addressing_mode is undefined or "dhcp" not in addressing_mode
 
 # TODO(TheJulia): Deprecation removal of the old network_info file name
 # should likely take place after an elongated deprecation cycle.
 # Begin deprecated in the Ocata release suggests Queens as the point
 # in which we should remove it.
-- name: "Place deprecated network info file location in each openstack/latest folder"
+- name: "Place deprecated network info file location in each openstack/<version> folder"
   template:
     src: network_data.json.j2
-    dest: "{{ variable_configdrive_location.stdout }}/{{ uuid }}/openstack/latest/network_info.json"
-  when: addressing_mode is undefined or "dhcp" not in addressing_mode
+    dest: "{{ variable_configdrive_location.stdout }}/{{ uuid }}/openstack/{{ item }}/network_info.json"
+  with_items: "{{ metadata_versions }}"
+  when:
+    - item in metadata_versions_supporting_network_data
```

Comment:
Shouldn't this predicate be inverted - 'item not in metadata_versions_supporting_network_data'?  Presumably network_data is sufficient in versions that support it.

Reply:
In the original code, both network_info and network_data were added to latest, not 2012-08-10. My change is intended to add 2015-10-15, and add them both to that also.

Reply:
OK thanks for clarifying.  Was the original code correct when it didn't put network_info.json into the older version dir?

Reply:
Yes - network_data  was introduced in liberty (2015). network_info was just another name for the same file that was used at one point. I'm not sure on the exact history but network_data is the correct name.